app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,346 @@
+            }
+        }
+    }
+}
+
+/**
+ * Text header for sections of synced tabs
+ *
+ * @param deviceName The name of the user's device connected that has synced tabs.
+ */
+@Composable
+fun SyncedTabsDeviceItem(deviceName: String) {
+    PrimaryText(
+        text = deviceName,
+        fontSize = 12.sp,
+        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+        letterSpacing = 0.04.sp,
+        maxLines = 1,
+        modifier = Modifier
+            .padding(16.dp, 16.dp, 8.dp, 7.dp)
+            .fillMaxWidth()
+    )
+}
+
+/**
+ * Synced tab list item UI
+ *
+ * @param tabTitleText The tab's display text.
+ * @param url The tab's URL.
+ * @param onClick The click handler when this synced tab is clicked.
+ */
+@Composable
+fun SyncedTabsTabItem(tabTitleText: String, url: String, onClick: () -> Unit) {
+    Column(
+        Modifier.clickable(
+            onClickLabel = tabTitleText,
+            onClick = onClick
+        )
+    ) {
+        PrimaryText(
+            text = tabTitleText,
+            fontSize = 16.sp,
+            maxLines = 1,
+            modifier = Modifier
+                .padding(16.dp, 0.dp)
+                .fillMaxWidth()
+        )
+        SecondaryText(
+            text = url,
+            fontSize = 12.sp,
+            maxLines = 1,
+            modifier = Modifier
+                .padding(PaddingValues(16.dp, 2.dp, 6.dp, 0.dp))
+                .fillMaxWidth()
+        )
+    }
+}
+
+/**
+ * Error UI to show if there is one of the errors outlined in [SyncedTabsView.ErrorType].
+ *
+ * @param errorType [SyncedTabsView.ErrorType] to be displayed after mapping to a string resource.
+ * @param navController [NavController] to call [NavGraphDirections.actionGlobalTurnOnSync()].
+ */
+@Composable
+fun SyncedTabsErrorItem(errorType: SyncedTabsView.ErrorType, navController: NavController? = null) {
+    val width: Float
+    val dashLength: Float
+    with(LocalDensity.current) {
+        width = 2.dp.toPx()
+        dashLength = 4.dp.toPx()
+    }
+    // As of December 2021, there is not an official way to get a rounded rectangle border with
+    // dashes using Compose. This is a next-best attempt with hard corners.
+    val stroke = Stroke(
+        width = width,
+        pathEffect = PathEffect.dashPathEffect(floatArrayOf(dashLength, dashLength), 0f)
+    )
+    val dashColor = FirefoxTheme.colors.borderDivider
+    Box(
+        Modifier
+            .padding(20.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        Canvas(Modifier.fillMaxSize()) {
+            drawRoundRect(color = dashColor, style = stroke)
+        }
+        Column(
+            modifier = Modifier
+                .padding(16.dp)
+                .fillMaxWidth()
+        ) {
+            PrimaryText(
+                text = stringResource(id = errorType.toStringRes()),
+                fontSize = 14.sp,
+                modifier = Modifier
+                    .padding(0.dp, 4.dp, 0.dp, 0.dp)
```

Comment:
```suggestion
                    .padding(top = 4.dp)
```
This can be simplify since the default values are 0.dp already.